robot/src/robot.py

```diff
@@ -18,9 +20,53 @@ def __init__(self):
 
         self.ultrasonic_sensor = UltrasonicSensor(Port.S4)
 
-    def obstacle_in_front(self):
+        self.routes = {1: [Event("MoveDistance", '{argument: 300}'), Event("Rotate", '{argument: 90}'),
+                           Event("MoveDistance", '{argument: 300}'), Event("Rotate", '{argument: 90}',
+                           Event("MoveDistance", '{argument: 300}'), Event("Rotate", '{argument: 90}'))]}
+
+        self.event_source = event_source
+
+
+    async def set_movement_speed(self, event):
+        speed = event.data.get('argument')
+        self.drive_base.settings(speed=speed)
+
+    # Get the distance traveled by the robot in centimeters
+    async def distance_travelled(self):
+        return self.drive_base.distance() / 10
+
+    async def obstacle_in_front(self):
         return self.ultrasonic_sensor.distance() > 300
 
+    async def stop_moving(self):
+        self.drive_base.stop()
+
+    # Direction == 1 is forward, direction == -1 is backward
+    async def move(self, event):
+        direction = event.data.get('argument')
+        if direction == 1:
+            self.drive_base.drive(200, 0)
+        elif direction == -1:
+            self.drive_base.drive(-200, 0)
+        else:
+            await self.stop_moving()
+
+    # Positive distance is forward, negative distance is backward
+    async def move_distance(self, event):
+        distance = event.data.get('argument')
+        self.drive_base.straight(distance)
+
+    # Positive angle is clockwise, negative angle is counterclockwise
+    async def rotate(self, event):
+        angle = event.data.get('argument')
```

Comment:
I'm just going to flag all of them, and then you can dismiss or fix depending on if it is correct python or not.